Game.Engine/wwwroot/js/settings.js

```diff
@@ -125,75 +107,42 @@ async function theme(v) {
         .then(text => {
             const info = JSON.parse(text);
 
-            var version = 1;
-            if (info.version) version = info.version;
+            const version = info.version || 1;
+
+            if (info.spriteModeMap) {
+                for (let key in info.spriteModeMap) {
+                    const modeMap = info.spriteModeMap[key];
 
-            if (info.files) {
-                // old format info.json
-                info.files.forEach(element => {
-                    zip.file(`daudmod/${element[0]}.png`)
+                    for (const mapKey in modeMap) spriteModeMap[key][mapKey] = modeMap[mapKey];
+                }
+            }
+
+            if (info.textureMap) {
+                for (let key in info.textureMap) {
+                    const map = info.textureMap[key];
+
+                    for (const textureKey in map) {
+                        if (!textureMap[key]) textureMap[key] = {};
+
+                        textureMap[key][textureKey] = map[textureKey];
+                    }
+
+                    zip.file(`daudmod/${map.file}.png`)
                         .async("arraybuffer")
                         .then(ab => {
                             const arrayBufferView = new Uint8Array(ab);
-                            const blob = new Blob([arrayBufferView], { type: "image/jpeg" });
+                            const blob = new Blob([arrayBufferView], { type: "image/png" });
                             const urlCreator = window.URL || window.webkitURL;
                             const url = urlCreator.createObjectURL(blob);
 
-                            textureMap[element[0]].url = url;
-                            if (element[1]) {
-                                var scale = element[1];
-                                if (version == 1 && element[0].startsWith("ship")) scale = 0.03;
-
-                                if (scale) textureMap[element[0]].scale = scale;
-                            }
+                            textureMap[key].url = url;
 
                             if (window.Game && window.Game.cache) {
```

Comment:
this bit really needs to only be called one time. i fumbled with the zip bit some... needs a .afterClose() or equiv